reconcile/utils/saasherder.py

```diff
@@ -388,13 +388,42 @@ def _process_template(self, options):
                 return None, None
 
             if "IMAGE_TAG" not in consolidated_parameters:
-                template_parameters = template.get('parameters')
-                if template_parameters is not None:
-                    for template_parameter in template_parameters:
-                        if template_parameter['name'] == 'IMAGE_TAG':
-                            # add IMAGE_TAG only if it is required
-                            image_tag = commit_sha[:hash_length]
-                            consolidated_parameters['IMAGE_TAG'] = image_tag
+                for template_parameter in template.get("parameters", {}):
+                    if template_parameter["name"] == "IMAGE_TAG":
+                        # add IMAGE_TAG only if it is required
+                        image_tag = commit_sha[:hash_length]
+                        consolidated_parameters["IMAGE_TAG"] = image_tag
+
+            # Do this in a separate loop, since it relies on IMAGE_TAG already
+            # being calculated.
+            # This should almost never already be set in
+            # consolidated_parameters, but just in case...
+            if "REPO_DIGEST" not in consolidated_parameters:
+                for template_parameter in template.get("parameters", {}):
+                    if template_parameter["name"] == "REPO_DIGEST":
+                        try:
+                            logging.info("Generating REPO_DIGEST.")
+                            import docker
+                            docker_client = docker.from_env()
+                            reg_img = consolidated_parameters["REGISTRY_IMG"]
+                            img_tag = consolidated_parameters["IMAGE_TAG"]
+                            reg_data = docker_client.images.get_registry_data(
```

Comment:
We have a lib, already part of the qontract-reconcile, that can grab the digest:
```python
>>> from sretoolbox.container import Image
>>> 
>>> i = Image('quay.io/app-sre/qontract-reconcile:latest', username=None, password=None)
>>> i.url_digest
'quay.io/app-sre/qontract-reconcile@sha256:9d119edb4c906e013c1b72c97cdab3fd26b35bed96f9b436990f43b9dbf9780e'
```

Missing image:

```python
>>> i = Image('quay.io/app-sre/missing:latest', username=None, password=None)
>>> i.url_digest
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/apahim/git/sretoolbox/sretoolbox/container/image.py", line 336, in url_digest
    url_digest += f'/{self.image}@{self.digest}'
  File "/home/apahim/git/sretoolbox/sretoolbox/container/image.py", line 79, in digest
    manifest = self._get_manifest()
  File "/home/apahim/git/sretoolbox/sretoolbox/container/image.py", line 94, in _get_manifest
    return self._request_get(url)
  File "/home/apahim/git/sretoolbox/sretoolbox/utils/retry.py", line 28, in f_retry
    defaults to ()
  File "/home/apahim/git/sretoolbox/sretoolbox/utils/retry.py", line 23, in f_retry
    :type exceptions: tuple(Exception, ...), optional
  File "/home/apahim/git/sretoolbox/sretoolbox/container/image.py", line 312, in _request_get
    self._raise_for_status(response)
  File "/home/apahim/git/sretoolbox/sretoolbox/container/image.py", line 286, in _raise_for_status
    raise HTTPError(msg)
requests.exceptions.HTTPError: (401) UNAUTHORIZED, access to the requested resource is not authorized
```

Reply:
Not asking you to use it, just presenting you with an option that's already part of the repo :)

Reply:
Neat! And it doesn't require the `docker` package. I'll switch it.

Reply:
I'm assuming is assuming the registry is up. Whats the exception if the registry is down?

Reply:
`requests.exceptions.ConnectionError`

Reply:
This is done.